ipn/localapi/localapi.go

```diff
@@ -915,6 +915,15 @@ func (h *Handler) serveServeConfig(w http.ResponseWriter, r *http.Request) {
 			writeErrorJSON(w, fmt.Errorf("decoding config: %w", err))
 			return
 		}
+
+		// require a local admin when setting a path handler
+		// TODO: roll-up this Windows-specific check into either PermitWrite
+		// or a global admin escalation check.
+		if !shouldPermitServeConfigUpdate(runtime.GOOS, configIn, h) {
```

Comment:
this func name is kinda backwards. It's misleading now as if the func returning true means it's permitted ("should permit!", the reader reads) but really line 909 says that's not all of it.

Howa about something super verbose like `shouldDenyServeConfigForGOOSAndUserContext`?